Sources/edge-agent/Services/EdgeContainerService.swift

```diff
@@ -163,26 +164,90 @@ struct EdgeContainerService: Edge_Agent_Services_V1_EdgeContainerService.Service
                             "source": "mqueue",
                             "options": ["nosuid", "noexec", "nodev"],
                         ],
+                        // Mount /sys for hardware visibility (GPU, USB controllers, etc.)
+                        [
+                            "destination": "/sys",
+                            "type": "sysfs",
+                            "source": "sysfs",
+                            "options": ["nosuid", "noexec", "nodev", "ro"],
+                        ],
+                        // Mount /sys/fs/cgroup for device cgroup access
+                        [
+                            "destination": "/sys/fs/cgroup",
+                            "type": "cgroup",
+                            "source": "cgroup",
+                            "options": ["nosuid", "noexec", "nodev", "relatime", "ro"],
+                        ],
                     ],
                     "linux": [
+                        // Keep standard namespace isolation for security
                         "namespaces": [
                             ["type": "pid"],
                             ["type": "ipc"],
                             ["type": "uts"],
                             ["type": "mount"],
                         ],
                         "networkMode": "host",
+                        // Only add capabilities needed for hardware/device access
                         "capabilities": [
-                            "bounding": ["SYS_PTRACE"],
-                            "effective": ["SYS_PTRACE"],
-                            "inheritable": ["SYS_PTRACE"],
-                            "permitted": ["SYS_PTRACE"],
+                            "bounding": [
```

Comment:
to be honest all of this should be in a config file that the agent loads and watches and then any change in capability is a config change.